### PR TITLE
Add MAD mode (#493)

### PR DESCRIPTION
**Pull request by JingfeiPeng**
Used approach 2 in #493

### CODE REVIEW

File: smarts/core/sumo_traffic_simulation.py [hunk 1]
```diff
@@ -272,7 +274,15 @@ def _close_traci_and_pipes(self):
...
-        for vehicle_id in self._non_sumo_vehicle_ids.union(self._sumo_vehicle_ids):
+        vehicles_to_remove = None
+        if self._kybersim_mode:
+            vehicles_to_remove = self._non_sumo_vehicle_ids
+        else:
+            vehicles_to_remove = self._non_sumo_vehicle_ids.union(
+                self._sumo_vehicle_ids
+            )
+
+        for vehicle_id in vehicles_to_remove:
```
Comment (JingfeiPeng) [inline smarts/core/sumo_traffic_simulation.py:285]: Could also change this and the above to the following if needed
```
(
            self._non_sumo_vehicle_ids
            if self._kybersim_mode
            else self._non_sumo_vehicle_ids.union(self._sumo_vehicle_ids)
        )
```

**Comment by liamchzh** [comment]
Nice! Actually, I was going to discuss the approaches in the parking log since we are introducing more toggles/params to meet the requirements, which I think should be fully considered before we get our hands dirty.

A few things:
* `allow_reload` and `kybersim_mode` are params customized for the KyberSim, should we keep the `allow_reload` or not
* If we decide to use separate params, then `kybersim_mode` is a bit too ambiguous
* We had talked about only exclude the vehicles with a specific prefix, how do we extend the API to support this

File: smarts/core/sumo_traffic_simulation.py [hunk 2]
```diff
@@ -66,7 +66,9 @@ def __init__(
         endless_traffic=True,
         allow_reload=True,
         debug=True,
+        kybersim_mode=False,
```
Comment (Gamenot) [inline smarts/core/sumo_traffic_simulation.py:69]: Use a different name.

File: smarts/core/sumo_traffic_simulation.py [hunk 3]
```diff
@@ -66,7 +66,9 @@ def __init__(
         endless_traffic=True,
         allow_reload=True,
         debug=True,
+        integration_mode=False,
     ):
```
Comment (Gamenot) [inline smarts/core/sumo_traffic_simulation.py:70]: We need a docstring here describing these parameters.

**Review by Gamenot — APPROVED** [review]

**Review by Adaickalavan — APPROVED** [review]

File: smarts/core/sumo_traffic_simulation.py [hunk 4]
```diff
@@ -54,6 +54,20 @@ class SumoTrafficSimulation:
...
+        auto_start:
+            False to pause simulation when SMARTS run, and wait for user to click 
+            start on sumo-gui
+        endless_traffic:
+            Not remove vehicles by re-adding back vehicles that exit 
+        allow_reload:
+            True to reload existing SUMO
+        remove_agents_only_mode: 
+            Remove only agent vehicles used by SMARTS and not delete other sumo
+            vehicles when teardown
```
Comment (JingfeiPeng) [inline smarts/core/sumo_traffic_simulation.py:70]: Feel free to point out anything inaccurate about the description

**Pull request merged**